File: 05-SimpleAdding/5.js
function SimpleAdding(num) { 
  var ans = 0;
  for(var i = num; i >= 0; i--){
    ans += i;
  }    
  
  return ans;
}

function SimpleAddingFun(num){
  if(num === 0){
    return 0
  }else{
    return num + SimpleAdding(num-1)
  }
}

console.log(SimpleAdding(10) == 55)
console.log(SimpleAdding(123))

File: 05-SimpleAdding/5.hs
simpleAdding :: Int -> Int
simpleAdding 0 = 0
simpleAdding x = x + simpleAdding (x-1)

simpleAddingTrivial :: Int -> Int
simpleAddingTrivial x = sum [1..x]

main = do
  print $ simpleAdding 10 == 55
  print $ simpleAdding 123